### ISSUE TITLE
SSH connection to Azure Linux VM with AAD auth not showing devicelogin token

### ISSUE THREAD

**Issue by rcbevans** [microsoft/vscode-remote-release#2538]
Hi,

I'm trying to connect to an Azure Ubuntu 18.04 VM with AAD Login enabled.

VSCode version 1.43.0,
Remote - SSH version 0.50.0
OpenSSH.Client version 0.0.1.0

When connecting using AAD username, the SSH process shows an https://aka.ms/devicelogin prompt and code which you are required to open up in a browser and complete to successfully log in, hitting enter after completing the device login.  See:

![image](https://user-images.githubusercontent.com/2041260/76582818-98388600-6494-11ea-95f7-23d8f88cc65b.png)

However, when attempting to connect using the Remote SSH extension in VSCode, the prompt with the device login code is not shown, making it impossible to connect.  See:

![image](https://user-images.githubusercontent.com/2041260/76583158-b05cd500-6495-11ea-8e98-10b15dbd582d.png)

It would be great if the extension could pick out that the ssh prompt is awaiting completion of a devicelogin requestion, giving the option to open up the browser with the code on the clipboard, or even better, complete the request for you in the background if you're logged in to the VSCode Azure extension with the account that's logging in, but at the very least the console output should contain the full askpass request content, with the devicelogin code for the user the be able to complete the login.

**Comment by roblourens** [comment]
Could you set these two settings and let me know whether this works?

```
"remote.SSH.useLocalServer": false,
"remote.SSH.showLoginTerminal": true,
```

It should pop open a terminal with the same prompt that you see in the terminal.

It's probably not possible for me to fix this nicely with the 'useLocalServer' default. Basically in that scenario I set the SSH_ASKPASS env var to some .cmd script, ssh invokes it with the auth prompt as the parameter, but multiline string params do not work well in cmd and only the first line is received. I haven't been able to find a solution for that.

**Comment by rcbevans** [comment]
Adding the two suggested settings works.  Can now see the full message including the devicelogin code in the terminal.

**Comment by rcbevans** [comment]
On Linux useLocalServer: false isn't necessary, it works with just showLoginTerminal.

**Comment by roblourens** [comment]
What did you see on Linux without those settings? I guess I would expect it to show a popup prompt with all of that text

**Comment by roblourens** [comment]
Oh I think you just edited - yeah you probably saw the whole thing on one line and possibly truncated, not a good experience.

**Comment by rcbevans** [comment]
Yes, edited to make it clearer which platform I was talking about in each case.

I was working with someone else using Linux, using Windows myself; he says it actually "just works" on Linux, the whole message does show up correctly in the confirmation box that pops up, including device code.

As you mentioned, looks like this is an issue specific to Windows due to cmd handling of the multiple lines.

**Comment by rcbevans** [comment]
The one issue he sees with it, is it's not possible to select the device login code from the dialog that's popping up, forcing him to manually type it out into the login page, or to fallback to using the terminal window rather than the prompt box to copy the code.

Would be great if the extension could be aware of this and make the navigation to devicelogin and copying of the code a little easier.

**Comment by rcbevans** [comment]
Ctrl+c copying the code from the terminal gets passed through the pipe and cancels the connection, but ctrl+shift+c/x allows copying the code from the terminal without cancelling the connection.

This is enough to get things unblocked for now.

**Comment by bpasham** [comment]
Quick workaround is provided [here on Stackoverflow](https://stackoverflow.com/questions/69573446/vscode-remote-ssh-does-not-work-with-ssh-using-azure-ad-authentication?noredirect=1#comment123083668_69573446). If anyone is interested
